File: lib/python/opentoken/src/main/opentoken/attributes/attribute_loader.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from typing import Set
from opentoken.attributes.attribute import Attribute
from opentoken.attributes.person.first_name_attribute import FirstNameAttribute
from opentoken.attributes.person.last_name_attribute import LastNameAttribute
from opentoken.attributes.person.birth_date_attribute import BirthDateAttribute
from opentoken.attributes.person.sex_attribute import SexAttribute
from opentoken.attributes.person.social_security_number_attribute import SocialSecurityNumberAttribute
from opentoken.attributes.person.postal_code_attribute import PostalCodeAttribute
from opentoken.attributes.person.age_attribute import AgeAttribute
from opentoken.attributes.person.birth_year_attribute import BirthYearAttribute
from opentoken.attributes.general.record_id_attribute import RecordIdAttribute
from opentoken.attributes.general.string_attribute import StringAttribute
from opentoken.attributes.general.date_attribute import DateAttribute
from opentoken.attributes.general.year_attribute import YearAttribute

class AttributeLoader:
    """
    Loads all available attribute implementations.
    """
    def __init__(self):
        raise RuntimeError("AttributeLoader should not be instantiated.")

    @staticmethod
    def load() -> Set[Attribute]:
        """Load all attribute implementations."""
        return {
            RecordIdAttribute(),
            StringAttribute(),
            DateAttribute(),
            YearAttribute(),
            FirstNameAttribute(),
            LastNameAttribute(),
            BirthDateAttribute(),
            AgeAttribute(),
            BirthYearAttribute(),
            SexAttribute(),
            SocialSecurityNumberAttribute(),
            PostalCodeAttribute()
        }

File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/AttributeLoader.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes;

import java.util.HashSet;
import java.util.ServiceLoader;
import java.util.Set;

public final class AttributeLoader {

    private AttributeLoader() {
    }

    public static Set<Attribute> load() {
        Set<Attribute> attributes = new HashSet<>();
        ServiceLoader<Attribute> loader = ServiceLoader.load(Attribute.class);
        for (Attribute attribute : loader) {
            attributes.add(attribute);
        }
        return attributes;
    }
}
